Add `derive(AsRefStr)` implementation

lo48576 opened the pull request:
* `derive(AsRefStr)` implements `AsRef<str>` (UPDATED: 558cdd0cd018dfe3a1b4fd8f8636ab514ac4cfd7)
* ~~`derive(AsStr)` uses `as_str`, `to_string`, `serialize` attributes (in this order).~~
* ~~`derive(ToString)` uses `to_string`, `as_str`, `serialize` attributes (in this order).~~
* `derive(AsRefStr)` uses `to_string` and `serialize` attributes (in this order, same as `derive(ToString)`). (UPDATED: a1b480dfc8c5e0bae9e07aed8821da4f1710383c, 558cdd0cd018dfe3a1b4fd8f8636ab514ac4cfd7)

I don't know which is better: `AsRef<str> for Enum` vs. `Enum::as_str()`, but I used `AsRef<str>` because it is similar to `ToString` (both are traits and therefore non-invasive).

Also I can't decide which name is better, `AsStr` vs. `AsRefStr`.

README documentation is not updated in this PR...

This solves #9.

Peternator7 commented on strum_macros/src/as_str.rs:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    if arms.len() < variants.len() {
+        arms.push(quote!{ _ => panic!("to_string() called on disabled variant.")})
```
Could you update this panic message for me?

lo48576 commented:
Ok, I'll rewrite the code to use `derive(AsRefStr)` and common `to_string` attribute.

lo48576 commented:
Removed `as_str` attribute and renamed `AsStr` to `AsRefStr`.

I'm sorry `to_string.rs` (https://github.com/Peternator7/strum/pull/10/files#diff-7eb81bb78680f1333200921a515022c5) has no content changes but trailing newline added...

Peternator7 commented:
This is merged. I'll get a new version deployed on crates.io hopefully tonight :)
